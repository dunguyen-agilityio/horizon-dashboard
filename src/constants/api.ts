export const API_ENTITY = {
  COMPLEX_REPORT: 'complex-report',
  CHECK_REPORT: 'check-report',
  FINANCIAL_REPORT: 'financial-report',
  USERS: 'users',
  SIGN_IN: 'auth/local',
  SIGN_UP: 'auth/local/register',
  FORGET_PASSWORD: 'auth/forgot-password',
  RESET_PASSWORD: 'auth/reset-password',
  CONTRIBUTOR: 'contributors',
};
