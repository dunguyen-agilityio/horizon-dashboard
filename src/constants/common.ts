export const QUERY_KEY = 'q';
export const STEP_KEY = 'step';
