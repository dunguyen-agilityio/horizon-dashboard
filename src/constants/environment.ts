export const API_ENDPOINT = process.env.API_ENDPOINT!;

// Firebase
export const FIREBASE_API_KEY = process.env.NEXT_PUBLIC_FIREBASE_API_KEY;
export const FIREBASE_AUTH_DOMAIN =
  process.env.NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN;
export const FIREBASE_PROJECT_ID = process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID;
export const FIREBASE_STORAGE_BUCKET =
  process.env.NEXT_PUBLIC_FIREBASE_STORAGE_BUCKET;
export const FIREBASE_SENDER_ID = process.env.NEXT_PUBLIC_FIREBASE_SENDER_ID;
export const FIREBASE_APP_ID = process.env.NEXT_PUBLIC_FIREBASE_APP_ID;
export const FIREBASE_MEASUREMENT =
  process.env.NEXT_PUBLIC_FIREBASE_MEASUREMENT;
export const FIREBASE_CLOUD_KEY = process.env.FIREBASE_CLOUD_KEY;
