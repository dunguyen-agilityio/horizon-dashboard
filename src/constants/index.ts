export * from './common';
export * from './images';
export * from './messages';
export * from './routes';
export * from './text';
export * from './endpoint';
export * from './session';
