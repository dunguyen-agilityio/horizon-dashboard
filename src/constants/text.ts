export const FOOTER_CONTENT =
  'Â© 2022 Horizon UI. All Rights Reserved. Made with love by Simple!';
