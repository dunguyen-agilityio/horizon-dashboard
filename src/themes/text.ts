export const fontSizes = {
  xs: '0.625rem', //10px
  tiny: '0.75rem', // 12px
  small: '0.875rem', // 14px
  medium: '1rem', // 16px
  large: '1.125rem', // 18px
  xl: '1.625rem', //  26px
  extra: '1.375rem', //22px
  '2xl': '2.125rem', // 34px
  '3xl': '2.5rem', //  40px
};

export const lineHeight = {
  tiny: '1rem', // 16px
  small: '1.25rem', //  20px
  medium: '1.5rem', //  24px
  large: '1.75rem', //  28px
  extra: '1.625px', //  26px
  '2xl': '2.625rem', // 42px
  xl: '2rem', // 32px
};
