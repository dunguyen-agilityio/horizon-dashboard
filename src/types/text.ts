export const enum TEXT_SIZE {
  '2xl' = '2xl',
  xl = 'xl',
  extra = 'extra',
  lg = 'lg',
  md = 'md',
  sm = 'sm',
  xs = 'xs',
}

export const enum TEXT_VARIANT {
  PRIMARY = 'primary',
  SECONDARY = 'secondary',
}
