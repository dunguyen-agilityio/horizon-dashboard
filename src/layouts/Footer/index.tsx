// Components
import { Text } from '@/components';
import { Link } from '@nextui-org/link';

// Types
import { TEXT_SIZE } from '@/types/text';

// Constants
import { FOOTER_ROUTES, FOOTER_CONTENT } from '@/constants';

const Footer = () => (
  <div className="flex flex-col justify-between items-center px-1 gap-2 lg:gap-0 lg:flex-row xl:px-7 mb-6">
    <Text className="text-secondary text-center" size={TEXT_SIZE.sm}>
      {FOOTER_CONTENT}
    </Text>
    <div className="flex gap-7 lg:gap-11">
      {FOOTER_ROUTES.map((footer) => (
        <Link href={footer.href} key={`${footer.title}`}>
          <Text size={TEXT_SIZE.sm} className="text-secondary leading-6">
            {footer.title}
          </Text>
        </Link>
      ))}
    </div>
  </div>
);

export default Footer;
