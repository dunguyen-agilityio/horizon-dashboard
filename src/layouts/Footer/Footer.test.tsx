import { render } from '@testing-library/react';

// Components
import Footer from '.';

// Constants
import { FOOTER_CONTENT } from '@/constants/text';

describe('Footer component', () => {
  it('Footer should render match snapshot', () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });

  it('Footer should render correct right content', () => {
    const { getByText } = render(<Footer />);

    expect(getByText(FOOTER_CONTENT)).toBeInTheDocument();
  });

  it('Footer should render correct left navigation item', () => {
    const { getByText } = render(<Footer />);

    expect(getByText('Marketplace')).toBeInTheDocument();
    expect(getByText('License')).toBeInTheDocument();
    expect(getByText('Terms of Use')).toBeInTheDocument();
    expect(getByText('Blog')).toBeInTheDocument();
  });
});
