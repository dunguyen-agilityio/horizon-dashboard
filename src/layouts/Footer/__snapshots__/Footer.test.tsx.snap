// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Footer should render match snapshot 1`] = `
<div>
  <footer
    class="sticky bottom-0 z-30 w-full bg-gray dark:bg-indigo-dark flex flex-col justify-between items-center px-1 gap-2 lg:gap-0 lg:flex-row xl:px-7 pb-6"
  >
    <p
      class="text-primary dark:text-white text-small font-medium text-secondary text-center"
    >
      Â© 2024 Horizon UI. All Rights Reserved. Made with love by Simple!
    </p>
    <div
      class="flex gap-7 lg:gap-11"
    >
      <a
        class="relative inline-flex items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-primary no-underline hover:opacity-80 active:opacity-disabled transition-opacity"
        href="/template"
        role="link"
        tabindex="0"
      >
        <p
          class="text-primary dark:text-white text-small font-medium text-secondary leading-6"
        >
          Marketplace
        </p>
      </a>
      <a
        class="relative inline-flex items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-primary no-underline hover:opacity-80 active:opacity-disabled transition-opacity"
        href="/license"
        role="link"
        tabindex="0"
      >
        <p
          class="text-primary dark:text-white text-small font-medium text-secondary leading-6"
        >
          License
        </p>
      </a>
      <a
        class="relative inline-flex items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-primary no-underline hover:opacity-80 active:opacity-disabled transition-opacity"
        href="/terms-of-use"
        role="link"
        tabindex="0"
      >
        <p
          class="text-primary dark:text-white text-small font-medium text-secondary leading-6"
        >
          Terms of Use
        </p>
      </a>
      <a
        class="relative inline-flex items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-primary no-underline hover:opacity-80 active:opacity-disabled transition-opacity"
        href="/blog"
        role="link"
        tabindex="0"
      >
        <p
          class="text-primary dark:text-white text-small font-medium text-secondary leading-6"
        >
          Blog
        </p>
      </a>
    </div>
  </footer>
</div>
`;
